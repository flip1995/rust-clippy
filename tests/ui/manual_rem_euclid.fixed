// run-rustfix

#![warn(clippy::manual_rem_euclid)]

fn main() {
    let value: i32 = 5;

    let _: i32 = value.rem_euclid(4);
    let _: i32 = value.rem_euclid(4);
    let _: i32 = value.rem_euclid(4);
    let _: i32 = value.rem_euclid(4);
    let _: i32 = 1 + value.rem_euclid(4);

    let _: i32 = (3 + value % 4) % 4;
    let _: i32 = (-4 + value % -4) % -4;
    let _: i32 = ((5 % 4) + 4) % 4;
}
