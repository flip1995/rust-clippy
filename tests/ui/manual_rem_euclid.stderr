error: manual `rem_euclid` implementation
  --> $DIR/manual_rem_euclid.rs:8:18
   |
LL |     let _: i32 = ((value % 4) + 4) % 4;
   |                  ^^^^^^^^^^^^^^^^^^^^^ help: consider using: `value.rem_euclid(4)`
   |
   = note: `-D clippy::manual-rem-euclid` implied by `-D warnings`

error: manual `rem_euclid` implementation
  --> $DIR/manual_rem_euclid.rs:9:18
   |
LL |     let _: i32 = (4 + (value % 4)) % 4;
   |                  ^^^^^^^^^^^^^^^^^^^^^ help: consider using: `value.rem_euclid(4)`

error: manual `rem_euclid` implementation
  --> $DIR/manual_rem_euclid.rs:10:18
   |
LL |     let _: i32 = (value % 4 + 4) % 4;
   |                  ^^^^^^^^^^^^^^^^^^^ help: consider using: `value.rem_euclid(4)`

error: manual `rem_euclid` implementation
  --> $DIR/manual_rem_euclid.rs:11:18
   |
LL |     let _: i32 = (4 + value % 4) % 4;
   |                  ^^^^^^^^^^^^^^^^^^^ help: consider using: `value.rem_euclid(4)`

error: manual `rem_euclid` implementation
  --> $DIR/manual_rem_euclid.rs:12:22
   |
LL |     let _: i32 = 1 + (4 + value % 4) % 4;
   |                      ^^^^^^^^^^^^^^^^^^^ help: consider using: `value.rem_euclid(4)`

error: manual `rem_euclid` implementation
  --> $DIR/manual_rem_euclid.rs:32:5
   |
LL |     ((num % 4) + 4) % 4
   |     ^^^^^^^^^^^^^^^^^^^ help: consider using: `num.rem_euclid(4)`

error: aborting due to 6 previous errors

